### DISCUSSION TITLE
C language interface

### DISCUSSION THREAD

**Discussion by superPN** [lldpd/lldpd#646] [Q&A]
Hi,
In the terminal, I can use "lldpcli show neighbors" to display the neighbor's information.
I want to retrieve LLDP neighbors' information in my C language program. Are there any C language interfaces available for this purpose? Which documents should I refer to?
I aim to acquire the number of LLDP neighbors, along with each neighbor's MAC address, Time to Live, System Capabilities, Enabled Capabilities, IPv4 Management Addresses, and some custom TLV information.

My config:
Ubuntu 20.04
I downloaded lldpd using "sudo apt-get install lldpd"
lldpd -v 1.0.4

Thanx for your work

**Comment by vincentbernat** [comment]
Check `src/lib/lldpctl.h`.

**Reply by superPN** [reply]
thank you for your reply

I'm trying to complete conformance testing of Ethernet/IP.There are some special requirements for LLDP in the Ethernet/IP protocol, but I don’t know much about LLDP.

My device is Ubuntu20.04, I used the "sudo apt-get install lldpd" command to download lldpd (1.0.4). I used "sudo apt-get install liblldpctl-dev" to download the lldpctl library.

I want to make some small modifications to the contents of the LLDP packet before it is sent, and write the contents to a file when the device receives LLDP packets from other devices.

I read the src/lib/lldpctl.h file and found the two callback functions "lldpctl_send_callback" and "lldpctl_recv_callback".

After I wrote my own callback function, I registered these two callback functions using the "lldpctl_new" function. When it ran, I found that it was not as I understood. The callback function is not called before sending the LLDP packet. See After taking a look at the default callback function "sync_send", the function of this callback function seems to be different from what I imagined.

I have no clue about this problem right now, can you give me an implementation idea and tell me how to implement this function?

Thank you very much again for your reply.

**Reply by vincentbernat** [reply]
The library is to query and configure lldpd. It cannot be used to receive or send LLDP packets.

**Reply by superPN** [reply]
thank you for your reply

I solved this problem by downloading the source code and making some modifications to the source code and then compiling and deploying lldpd.
I found that deploying lldpd in this way cannot find the configuration files "etc/lldpd.conf" and "etc/lldpd.d".

Do I need to add any options when executing the `./configure` command?Or is there any other way to add the `configuration configure lldp custom-tlv oui 00,21,6C subtype 1 oui-info CE,06,0C,00,E9 ,FD,02,03,15,CD,5B,07`?
